contracts/Core.sol

```diff
@@ -92,11 +112,88 @@ contract Core is CoreInterface {
 		return coreChainIdRemote;
 	}
 
+	/// @dev public function openSTRemote
+	/// @return address coreOpenSTRemote
 	function openSTRemote()
 		public
 		view
 		returns (address /* OpenSTRemote */)
 	{
 		return coreOpenSTRemote;
 	}
+
+	/**
+     * Commit new state root for a block height
+     *
+     */
+	/// @dev Commit new state root for a block height
+	/// @param _blockHeight block height for which stateRoots mapping needs to update
+	/// @param _stateRoot state root of input block height
+	/// @return bytes32 stateRoot
+	function commitStateRoot(
+		uint256 _blockHeight,
+		bytes32 _stateRoot)
+		external
+		returns(bytes32 stateRoot)
+	{
+		// check if the caller is whitelisted worker
+		require(workers.isWorker(msg.sender), "Invalid worker address");
+		// State root should be valid
+		require(_stateRoot != bytes32(0), "Invalid state root");
+		// Input block height should be valid
+		require(_blockHeight > latestStateRootBlockHeight, "Given block height is lower or equal to highest committed state root block height.");
+
+		stateRoots[_blockHeight] = _stateRoot;
+		latestStateRootBlockHeight = _blockHeight;
+
+		emit StateRootCommitted(_blockHeight, _stateRoot);
+
+		return stateRoot;
+	}
+
+	/// @dev Verify account proof of OpenSTRemote and commit storage root at given block height
+	/// @param _blockHeight block height at which OpenST is to be proven
+    /// @param _rlpEncodedAccount rlpencoded account node object
+    /// @param _rlpParentNodes RLP encoded value of account proof parent nodes
+    /// @return bool status
+	function proveOpenST(
+		uint256 _blockHeight,
+		bytes _rlpEncodedAccount,
+		bytes _rlpParentNodes)
+		external
+		returns(bool status)
+	{
+		// check if the caller is whitelisted worker
+		require(workers.isWorker(msg.sender), "Invalid worker address");
+		// Check for block height
+		require(_blockHeight != 0, "Invalid block height");
+		// Storage root should be valid
+		require(_rlpEncodedAccount.length != 0, "Invalid rlp encoded account value");
+
+		bytes32 stateRoot = stateRoots[_blockHeight];
+		// State root should be present for the block height
+		require(stateRoot != bytes32(0), "State root missing for given block height");
+
+		// Decode RLP encoded account value
+		RLP.RLPItem memory accountItem = RLP.toRLPItem(_rlpEncodedAccount);
+		// Convert to list
+		RLP.RLPItem[] memory accountArray = RLP.toList(accountItem);
+		// Array 3rd position is storage root
+		bytes32 storageRoot = RLP.toBytes32(accountArray[2]);
+		// Hash the rlpEncodedValue value
+		bytes32 hashedAccount = keccak256(_rlpEncodedAccount);
+
```

Comment:
We also need to emit event when account proof is verified since platform will be relying on this.
For event we need _blockHeight, storageRoot and hashedAccount.

In case we need to by pass above code, we can get _blockHeight and storageRoot but not hashedAccount as it's calculated in the end. What do you suggest?